web-marketplace/src/components/organisms/ChooseCreditsForm/ChooseCreditsForm.tsx

```diff
@@ -73,230 +78,268 @@ export type Props = {
+      name: CURRENCY_AMOUNT,
+    });
 
-  const currency = useWatch({
-    control: form.control,
-    name: CURRENCY,
-  });
-  const creditsAmount = useWatch({
-    control: form.control,
-    name: CREDITS_AMOUNT,
-  });
-  const currencyAmount = useWatch({
-    control: form.control,
-    name: CURRENCY_AMOUNT,
-  });
+    useEffect(() => {
+      if (currency.askDenom !== USD_DENOM) {
+        form.trigger(CURRENCY_AMOUNT);
+      }
+    }, [userBalance, form, currency.askDenom]);
 
-  const filteredCryptoSellOrders = useMemo(
-    () =>
-      getFilteredCryptoSellOrders({
-        askDenom: currency?.askDenom,
-        cryptoSellOrders,
-        retiring,
-      }),
-    [cryptoSellOrders, currency?.askDenom, retiring],
-  );
+    const filteredCryptoSellOrders = useMemo(
+      () =>
+        getFilteredCryptoSellOrders({
+          askDenom: currency?.askDenom,
+          cryptoSellOrders,
+          retiring,
+        }),
+      [cryptoSellOrders, currency?.askDenom, retiring],
+    );
 
-  const orderedSellOrders = useMemo(
-    () =>
-      card
-        ? cardSellOrders.sort((a, b) => a.usdPrice - b.usdPrice)
-        : filteredCryptoSellOrders?.sort(
-            (a, b) => Number(a.askAmount) - Number(b.askAmount),
-          ) || [],
+    const orderedSellOrders = useMemo(
+      () =>
+        card
+          ? cardSellOrders.sort((a, b) => a.usdPrice - b.usdPrice)
+          : filteredCryptoSellOrders?.sort(
+              (a, b) => Number(a.askAmount) - Number(b.askAmount),
+            ) || [],
 
-    [card, cardSellOrders, filteredCryptoSellOrders],
-  );
+      [card, cardSellOrders, filteredCryptoSellOrders],
+    );
 
-  const handleCryptoPurchaseOptions = useCallback(() => {
-    setRetiring(prev => !prev);
-  }, [setRetiring]);
+    const handleCryptoPurchaseOptions = useCallback(() => {
+      setRetiring(prev => !prev);
+    }, [setRetiring]);
 
-  const handlePaymentOptions = useCallback(
-    (option: string) => {
-      setPaymentOption(option as PaymentOptionsType);
-      form.setValue(CREDIT_VINTAGE_OPTIONS, []);
-      form.setValue(
-        CURRENCY,
-        option === PAYMENT_OPTIONS.CARD ? cardCurrency : defaultCryptoCurrency,
-      );
-    },
-    [setPaymentOption, form, cardCurrency, defaultCryptoCurrency],
-  );
+    const handlePaymentOptions = useCallback(
+      (option: string) => {
+        startTransition(() => {
+          setPaymentOption(option as PaymentOptionsType);
+        });
+        form.reset({
+          ...form.getValues(),
+          [CREDIT_VINTAGE_OPTIONS]: [],
+          [CURRENCY]:
+            option === PAYMENT_OPTIONS.CARD
+              ? cardCurrency
+              : defaultCryptoCurrency,
+        });
+        const currency =
+          option !== PAYMENT_OPTIONS.CARD
+            ? { askDenom: REGEN_DENOM, askBaseDenom: REGEN_DENOM }
+            : { askDenom: USD_DENOM, askBaseDenom: USD_DENOM };
+        updateMultiStepCurrency(handleSave, data, currency, activeStep);
```

Comment:
why are we setting different values for currency here?
it could even be that REGEN is not available as askDenom